WeakAuras/BossMods.lua

```diff
@@ -0,0 +1,1598 @@
+  statesParameter = "one",
+  canHaveDuration = "timed",
+  delayEvents = true,
+  timedrequired = true
+}
+Private.category_event_prototype.addons["Boss Mod Stage (Event)"] = L["Boss Mod Stage (Event)"]
+
+Private.event_prototypes["Boss Mod Announce"] = {
+  type = "addons",
+  events = {},
+  internal_events = {
+    "BossMod_Announce"
+  },
+  name = L["Boss Mod Announce"],
+  init = function(trigger)
+    Private.ExecEnv.BossMods.Generic:RegisterMessage();
+    local ret = "local use_cloneId = %s;"
+    return ret:format(trigger.use_cloneId and "true" or "false");
+  end,
+  statesParameter = "all",
+  args = {
+    {
+      name = "spellId",
+      init = "arg",
+      display = L["Key"],
+      type = "string",
+      store = true,
+      conditionType = "string"
+    },
+    {
+      name = "message",
+      init = "arg",
+      display = L["Message"],
+      type = "longstring",
+      store = true,
+      conditionType = "string"
+    },
+    {
+      name = "name",
+      init = "message",
+      hidden = true,
+      test = "true",
+      store = true,
+    },
+    {
+      name = "icon",
+      init = "arg",
+      store = true,
+      hidden = true,
+      test = "true"
+    },
+    {
+      name = "cloneId",
+      display = L["Clone per Event"],
+      type = "toggle",
+      test = "true",
+      init = "use_cloneId and WeakAuras.GetUniqueCloneId() or ''"
+    },
+    {
+      name = "note",
+      type = "description",
+      display = "",
+      text = function()
+        local ret = L["Note: This trigger will use BigWigs or DBM, in that order if both are installed.\n\nActive boss mod addon: "]
+        if Private.ExecEnv.BossMods.BigWigs.isInstalled then
+          ret = ret .. "BigWigs"
+        elseif Private.ExecEnv.BossMods.DBM.isInstalled then
+          ret = ret .. "DBM"
+        else
+          ret = ret .. L["none detected"]
+        end
+        return ret
+      end
+    },
+  },
+  timedrequired = true
+}
+Private.category_event_prototype.addons["Boss Mod Announce"] = L["Boss Mod Announce"]
+
+Private.event_prototypes["Boss Mod Timer"] = {
+  type = "addons",
+  events = {},
+  internal_events = {
+    "BossMod_TimerStart", "BossMod_TimerStop", "BossMod_TimerUpdate", "BossMod_TimerForce", "BossMod_TimerResume", "BossMod_TimerPause"
+  },
+  force_events = "BossMod_TimerForce",
+  name = L["Boss Mod Timer"],
+  canHaveDuration = "timed",
+  triggerFunction = function(trigger)
+    Private.ExecEnv.BossMods.Generic:RegisterTimer()
+    local ret = [=[
+      local triggerCounter = %q
+      local counter
+      if triggerCounter and triggerCounter ~= "" then
+        counter = Private.ExecEnv.CreateTriggerCounter(triggerCounter)
+      else
+        counter = Private.ExecEnv.CreateTriggerCounter()
+      end
+      local isBW = Private.ExecEnv.BossMods.Generic == Private.ExecEnv.BossMods.BigWigs
+      local isDBM = Private.ExecEnv.BossMods.Generic == Private.ExecEnv.BossMods.DBM
+      local extraArg1, extraArg2
```

Comment:
I dislike that the generic trigger has to check which of the boss mods is active.

I'd much prefer if the BossMods "modules" would provide a function, e.g. CooldownTimerMatches(...), that behaved in the desired way and had the same api for both dbm and bw. So that the trigger has to not care about any differences.